### PR TITLE
Raise when trying to acquire in R/O or missing folder

### PR DESCRIPTION
**Pull request by gaborbernat**
Signed-off-by: Bernát Gábor <gaborjbernat@gmail.com>

### CODE REVIEW

File: src/filelock/_soft.py [hunk 1]
```diff
@@ -10,8 +14,11 @@ def _acquire(self):
         open_mode = os.O_WRONLY | os.O_CREAT | os.O_EXCL | os.O_TRUNC
         try:
             fd = os.open(self._lock_file, open_mode)
-        except OSError:
-            pass
+        except OSError as exception:
+            if exception.errno in (EPERM, EACCES, ENOENT):
+                raise
+            if sys.platform == "win32" and not os.access(self._lock_file, os.W_OK):
+                raise PermissionError("Permission denied")
```
Comment (FFY00) [inline src/filelock/_soft.py]: Maybe `raise ... from None` to remove the exception chaining.

Comment (gaborbernat) [reply src/filelock/_soft.py]: that doesn't work on python 2, that we still support here; perhaps once we drop it

Comment (FFY00) [reply src/filelock/_soft.py]: Ah, of course.

**Comment by gaborbernat** [comment]
I'm not sure we can make the soft file lock deadlock safe 🤦 but will keep trying 😟

**Review by gaborbernat — COMMENTED** [review]
The end result feels hacky, but hey it works 🤷

**Pull request merged**